src/main/interface/msp.c

```diff
@@ -870,25 +872,30 @@ bool mspProcessOutCommand(uint8_t cmdMSP, sbuf_t *dst)
 
     case MSP_NAME:
         {
-            // Show warning for DJI OSD instead of pilot name if osd_warning_enabled and usb not connected
-            if (!usbVcpIsConnected() && osdWarnGetState(OSD_WARNING_DJI)) {
-                const char *warning = pilotConfig()->warning;
-                const int len = strlen(warning);
-                for (int i = 0; i < len; i++) {
-                	// skip non printable chars
-                    if (isprint(warning[i])) {
-                        sbufWriteU8(dst, warning[i]);
-                    } else {
-                        sbufWriteU8(dst, 0);
+            // Show warning for DJI OSD instead of pilot name if osd_warning_enabled
+            if (osdWarnGetState(OSD_WARNING_DJI)) {
+                bool isUsbConnected = false;
+#ifdef USE_VCP
+                isUsbConnected = usbVcpIsConnected();
+#endif
+                if (!isUsbConnected) {
+                    unsigned int len = sizeof(pilotConfig()->warning);
+                    for (unsigned int i = 0; i < len; i++) {
+                        // skip non printable chars
+                        if (isprint(pilotConfig()->warning[i])) {
+                            sbufWriteU8(dst, pilotConfig()->warning[i]);
+                        } else {
+                            sbufWriteU8(dst, 0);
+                        }
                     }
+                    break;
```

Comment:
as i understand it the break here prevent to go further for the case dji enable  && not Vcp and connected to usb @mbilozub ?

Reply:
Yes, because otherwise, you will see a warning as a craft name, such as BATT < FULL or * * * * when you use the EmuflightCnfigurator.

Reply:
ok good !